static/resources/harness.js

```diff
@@ -764,50 +764,70 @@
     resultsEl.appendChild(resultEl);
   }
 
-  function report(results, hideResults) {
-    updateStatus('Tests complete. Posting results to server...');
+  function sendReport(results) {
+    var body = JSON.stringify(results);
 
-    try {
-      var body = JSON.stringify(results);
+    var client;
+    if ('XMLHttpRequest' in self) {
+      client = new XMLHttpRequest();
+    } else if ('ActiveXObject' in self) {
+      client = new ActiveXObject('Microsoft.XMLHTTP');
+    }
 
-      var client;
-      if ('XMLHttpRequest' in self) {
-        client = new XMLHttpRequest();
-      } else if ('ActiveXObject' in self) {
-        client = new ActiveXObject('Microsoft.XMLHTTP');
+    if (!client) {
+      updateStatus(
+        'Cannot upload results: XMLHttpRequest is not supported.',
+        'error-notice'
+      );
+      return;
+    }
+
+    var resultsURL =
+      (location.origin || location.protocol + '//' + location.host) +
+      '/api/results?for=' +
+      encodeURIComponent(location.href);
+
+    client.open('POST', resultsURL);
+    client.setRequestHeader('Content-Type', 'application/json;charset=UTF-8');
+    client.send(body);
+    client.onreadystatechange = function () {
+      if (client.readyState == 4) {
+        if (client.status >= 200 && client.status <= 299) {
+          document.getElementById('export-download').disabled = false;
+          document.getElementById('export-github').disabled = false;
+          updateStatus('Results uploaded.', 'success-notice');
+        } else {
+          updateStatus(
+            'Failed to upload results: server error.',
+            'error-notice'
+          );
+          consoleLog('Server response: ' + client.response);
+        }
       }
+    };
+  }
 
-      if (!client) {
-        updateStatus(
-          'Cannot upload results: XMLHttpRequest is not supported.',
-          'error-notice'
+  function report(results, hideResults) {
+    updateStatus('Tests complete. Posting results to server...');
+
+    try {
+      if ('JSON' in self && 'parse' in JSON) {
+        sendReport(results);
+      } else {
+        // Load JSON polyfill if needed
+        var polyfill = document.createElement('script');
+        polyfill.src = '/resources/json3.min.js';
+
+        polyfill.addEventListener(
```

Comment:
Does this work as far back as the previous code? `addEventListener` was added in IE9, right?

Reply:
I've double-checked to make sure this doesn't cause an issue in older browsers  -- it functions in IE8, though it seems IE6 and IE7 are having a little issue.  Looking into it now!

Reply:
Alright, looks like IE doesn't support the `load` event before IE9, so I've added a workaround where the report will be sent after a small delay instead.  Fortunately this is an IE-only issue, other browsers aren't affected!